### PR TITLE
Kill event system

### PR DESCRIPTION
**Pull request by shua**
States don't send signals to themselves, deleted commented Event stuff. Test framework works the same as before I changed the states.

Should merge cleanly. Just something I thought was a little messy, this hopefully made it better.

### CODE REVIEW

File: engine/src/Scene/State.cpp [hunk 1]
```diff
@@ -55,6 +55,11 @@ void State::DeleteScene(const QString& name) {
 }
 
 void State::UpdateFrame(double simulation_frame_time) {
+    UpdateSceneFrame(simulation_frame_time);
+    UpdateStateFrame(simulation_frame_time);
```
Comment (svenstaro) [inline engine/src/Scene/State.cpp:59]: Why twice?

**Comment by svenstaro** [comment]
A few issues:
- I don't see docs on the new renamed/methods
- It doesn't compile: http://pastie.org/2680862
- It doesn't merge cleanly

File: engine/src/Scene/State.cpp (hunk 1 above)
Comment (shua) [inline engine/src/Scene/State.cpp:59]: One updates the scenes, the other is updates  the state and is virtual, and defined in the samples/tests/whatever inherits from State. A couple lines up, it used to connect the BeginFrame signal to a slot in every scene, but someone put in a manual function that just iterates through all the scenes and updates them. I figured I'd just use that, and this way, we don't have to call connect(this, SIGNAL(BeginFrame(double)), this, SLOT(_HandleEvent(double)), which is wasteful.

**Comment by shua** [comment]
yeah, there were a couple new commits before I pull requested, I'll check on it, and add docs.

**Comment by shua** [comment]
all network stuff still broken at run, but compile fine.

**Comment by shua** [comment]
Still have to fix merge, my bad.

**Pull request merged**